### ISSUE TITLE
OccupancyStatus vs. occupancy_percentage

### ISSUE THREAD

**Issue by barbeau** [google/transit#223]
In GTFS-RT, we now have two experimental fields that can contain real-time vehicle occupancy information:
* [OccupancyStatus](https://github.com/google/transit/blob/master/gtfs-realtime/spec/en/reference.md#enum-occupancystatus) - A enum of 7 values added to the spec 6 years ago. We know of 3 producers of this data - see [this pull request](https://github.com/google/transit/pull/212) for details.
* [occupancy_percentage](https://github.com/google/transit/blob/master/gtfs-realtime/spec/en/reference.md#message-vehicleposition) - A percentage value representing the degree of passenger occupancy of the vehicle. Just adopted as an experimental field ~30 days ago in [this pull request](https://github.com/google/transit/pull/213)

Unless there is a specific use case to have both fields in the spec, we should try to narrow down to one.

Producers and consumers - if you were to pick **one** of the two, which would you pick and why?

For producers - is one field easier to produce than the other? Would adoption be faster with one field in particular?

**Comment by skinkie** [comment]
If it would be one or the other; To be compatible with SIRI I would pick the enumeration. In in addition, it is the least commercial sensitive information to be shared.

**Comment by darylweinberg** [comment]
At Capital Metro we are working to produce OccupanyStatus by August. It will not have a percentage value at this time.

**Comment by stevenmwhite** [comment]
`occupancy_percentage` would be the pick.
- Lends itself better to situational adjustments (for example during the COVIC crisis, 100% could represent a policy-related limit rather than a physical limit). It makes sense as "per agency regulations (whatever they may be at a given time), the vehicle is X percent of its limit. The scale of 100% can change as needed.
- It is less open to interpretation (what is many, what is few?) by different agencies and different consumers.
- It does not claim precession that may not be accurate (when status says a bus is standing room only, it may not necessarily be... depending on whether passengers have chosen to take all available seats before standing. It's not uncommon to be on a crowded transit vehicle with many people standing and still have some open seats.)
- It is easier to produce and I believe adoption would be faster as a result.
- Gives consumers more flexibility in how they display it, rather than attempting to map the enums to approximate percentage levels.

What it doesn't do is give any sort of accessibility information. While `occupancyStatus` can be unclear, the mention of "seats" can be helpful to riders if it is accurate, and a pure percentage doesn't give that.

**Comment by mike-swiftly** [comment]
I don't believe these fields are mutually exclusive. Both would be optional, so clients will be able to handle data being available in none, one, or both of the fields. I expect that more agencies will be able to handle the enumeration because it is less information. But really don't want to preclude applications that can use more detail, like occupancy percentage.

**Comment by stevenmwhite** [comment]
> I don't believe these fields are mutually exclusive. Both would be optional, so clients will be able to handle data being available in none, one, or both of the fields.

I agree. I wouldn't necessarily advocate for deprecating one. My comments above should be taken in the context of if we had to pick one or the other.

**Comment by mcossette1** [comment]
> > I don't believe these fields are mutually exclusive. Both would be optional, so clients will be able to handle data being available in none, one, or both of the fields.
> 
> I agree. I wouldn't necessarily advocate for deprecating one. My comments above should be taken in the context of if we had to pick one or the other.

I agree too.

**Comment by BodoMinea** [comment]
I agree with most of the points above. While this is probably not the outcome desired by feed spec maintainers, as I understand that simplicity is key, I think that both of these fields could be of use to some.

While the enumeration is simplified enough to not give away data of commercial value, for some agencies that are just starting to implement this system, a percentage may be easier to implement, by averaging the data from sensors for example - vehicle cabin weight scales or volumetric cameras.

I also think that applications that consume data may have interest in having both fields exposed - the enumeration is easy to implement as a simple color highlight/people icon, the percentage may be easier to manipulate in a mathematics formula for weighting routing options and also for presenting more exact data to the user.

I would be fine with both of them being optional and given the available resources at a client, I would do my best to provide both.

**Comment by antrim** [comment]
If we keep both of these fields, it might create opportunities for confusion or unnecessary spec bloat. I worry we’d take a situation with existing ambiguity and just layer in more ambiguity and complexity:
- How should a consumer interpret a feed where occupancy_percentage is `90` and occupancy_status is `FULL`?
- Should there be a consistent mapping applied between occupancy_status and occupancy_percentage?
- If so, why not just have one field?

As part of this exercise, I ask what would be lost (if anything) by just having one field?

**Comment by skinkie** [comment]
@antrim I really liked how @gcamp has defined the percentage based on the safety limit of the vehicle. For me the enumeration would be just the textual description, presented to as translatable text to the end user. It would obviously be nice to have a ballpark as mapping to have a GTFS-RT validator to invalidate 10% to FULL.

Our percentage / enum approach is mainly targeting busses and taxi's. But for a train with different compartments the values greatly variate. We have not modeled that yet. Neither did we do the mobility impaired space, which is even deeper level of detail than the current two approaches. I think after the VehicleTypes GTFS stuff, more questions will come done. Some train builders are already able to tell exact numbers of free seats. Similar to the number of people in the standing space.

**Comment by juanborre** [comment]
Another point for the enum is that it is the only way a producer can specify if a passenger is likely to find a seat. The mapping between percentage and enum will greatly depend on the vehicle and its seating capacity.

Getting rid of that field would remove that functionality. 

However, if the only reason for the enum was to add information about seating, it could be replaced with another field to address that need in particular.

**Comment by caywood** [comment]
@barbeau this strikes me as a false choice resulting from the historical evolution of the spec rather than today's crisis.  Given there are few actual implementations of the spec we should choose whatever serves today's needs best. Apologies if you've already considered the below alternative 

Within GTFS-RT VehiclePositions:
- passenger_count_estimate
- seating_capacity
- max_capacity

seating_capacity is a physical property of the vehicle

max_capacity is a policy decision - normally >= seating_capacity but in COVID, reduced to whatever the agency says the max is.

This would allow consumers to reconstruct any OccupancyStatus or occupancy_percentage they care to represent, with greater precision if needed.

And any producers currently producing OccupancyStatus with APCs must already know seating_capacity. Are there any technologies being used that can only produce a percentage?

(seating_capacity and max_capacity are already part of [GTFS-vehicles full](https://docs.google.com/document/d/156RiBjI6FnWJvO8_XWX11Q9nLdOiBdS_rilA-oamlv8/edit#heading=h.tosuo6e9e0z7) but I don't expect that will be implemented on the relevant timeline for the current problem - we need a solution within GTFS-RT VehiclePositions.

**Comment by skinkie** [comment]
@caywood I agree on the passenger_count_estimate (and relevant mobility impaired information) but having static data in a realtime data stream, because we fail to standardise the static data is a direction I don't wish to follow either.

**Comment by barbeau** [comment]
@caywood Thanks for the feedback! Sure, additional ideas are welcome too if the current fields don't meet certain use cases. Our goal is to get to something that meets the producers and consumers needs.

Do any other producers have comments on the fields proposed in https://github.com/google/transit/issues/223#issuecomment-627099138, vs the existing experimental occupancy fields in GTFS-RT?

**Comment by mike-swiftly** [comment]
I believe that seating_capacity and max_capacity should not be included in the real-time feed because as @skinkie stated, that is a static data issue. Even if it were to be added to the real-time spec would first need to figure out how to define it. I think the first question is whether it should it go into GTFS.

**Comment by skinkie** [comment]
@mike-swiftly I think a proper discussion should follow about a standard that is far more low level than travel information. I have not yet reviewed ITxPT, but that might have defined such exchanges.

**Comment by mike-swiftly** [comment]
Here is another topic for discussion with respect to occupancyStatus and occupancy_percentage: **should they also be included with predictions in the TripUpdates feed?** 

The benefit would be that a client, like an electronic sign system, would then not need to read both TripUpdate and VehiclePositions feed and combine the info. Instead, it would be very easy to display a prediction like "5 minutes - Many Seats Available".

A counter argument could be that displaying vehicle count when a bus is 20 minutes away is irrelevant, that the count could change greatly by the time the bus arrives. But I think that the client should decide what information to display, and that the real-time information system should simply provide possibly useful information to the client.

**Comment by skinkie** [comment]
> But I think that the client should decide what information to display, and that the real-time information system should simply provide possibly useful information to the client. 

You basically suggest application specific information profiles, which SIRI uses extensively.

**Comment by caywood** [comment]
@mike-swiftly My only concern about adding seating_capacity and max_capacity to GTFS would be whether agencies know capacity of the vehicle serving each trip ahead of time. If you (and other transit operations experts) say that would work, I agree it would be better in GTFS as static data. 

PS if the primary concern is how easy is it to write a client - then we should definitely put seating_capacity and max_capacity in the VehiclePositions feed so consumers don't have to merge GTFS with GTFS-RT VehiclePositions! Also let's get rid of protocol buffers :)

**Comment by barbeau** [comment]
>Here is another topic for discussion with respect to occupancyStatus and occupancy_percentage: should they also be included with predictions in the TripUpdates feed?

@mike-swiftly I'd prefer to have TripUpdates specifically contain **predictive** occupancy information within StopTimeUpdates. As you imply, if an agency knows historical on/offs at specific stops, the current occupancy could likely differ significantly from the expected occupancy in a few stops (e.g., stops near the end of the line).

Similarly, I'd prefer to see observed schedule_deviation (how the vehicle's departure time compared to the schedule at the last stop it passed) be stored in VehiclePositions and TripUpdates contain only predictions (what the vehicle is expected to do in the future).

IMHO consumers would benefit from a clearer delineation between observed facts about the current state of the system and predictions about what may happen in the future.

**Comment by caywood** [comment]
Updated proposal based on feedback from @mike-swiftly and @skinkie

Add to GTFS-RT VehiclePosition:
    passenger_count_estimate - best available estimate of current number of passengers, but agencies can use a discrete set of values or add random noise if privacy sensitive.

Add to GTFS trips.txt
- seating_capacity (physical property of vehicle)
- max_capacity (allowable crowding based on policy decision)

(note trips.txt already includes wheelchair_accessible and bikes_allowed which are analogous vehicle properties)

**Comment by darylweinberg** [comment]
I understand the value of seating_capacity as a physical property of the vehicle in GTFS trips.txt but we do not use the same series of vehicle to run the same trip every day. Some blocks will have restrictions on vehicle size so those will stay fairly consistent but others will not.

As an example, our Rapid service is operated by both 60 ft articulated buses and 40 ft and those assignments vary. There have also been times that we have not had enough vehicles available and have had to supplement the Rapid fleet with with other 35 ft or 40 ft buses.

As such, seating_capacity in trips.txt would be more of a guideline than actual seating capacity for a given trip on a given day.

**Comment by skinkie** [comment]
@darylweinberg I would envision the option to communicate the Vehicle (and thus VehicleType) in realtime data. Hence overriding the values provided in the schedule.

**Comment by skinkie** [comment]
> Add to GTFS trips.txt

Would you really want to add such this to trips, opposed to a VehicleType for example?

**Comment by stevenmwhite** [comment]
In our experience it’s not necessarily common for vehicles of the exact same capacity to run the same trips regularly. Even if it’s the same physical size vehicle (which it’s not always), most of our agency customers have fleets with a wide range of years and different model years have slightly different seating capacities and total capacities.

Other notes about this new concept of giving capacity and actual and then letting consumers figure out the percentage...
- I like it in theory, but there was a clear directive in the proposed spec for percentage that said you should not be able to track a single person boarding/alighting.
- I think that offering a field for passenger count makes this privacy much less likely in practice (even with the note about added noise or discrete values, the easiest way to implement it to just report what you have from APC systems).
- I’m not fully against it but am curious to hear if anyone else has concerns about that aspect?

**Comment by caywood** [comment]
@stevenmwhite the potential use of different vehicles is why I originally put capacity in GTFS-RT rather than trips.txt. If we need to allow realtime overrides - is it still useful to have it in the GTFS trips.txt? 

I'm trying to understand the privacy concern but maybe I'm missing something. Except in the very rare case where you have 0 passengers and then go to 1 then back to 0, you can't identify an individual passenger. (The [proposal for occupancy_percentage](https://github.com/google/transit/blob/master/gtfs-realtime/spec/en/reference.md#message-vehicleposition) had debate without clear guidance on how to preserve privacy either - if you have 2.5% occupancy of a 40 seat vehicle, you know what the count is). @gcamp can you explain what you were trying to accomplish for privacy?

**Comment by stevenmwhite** [comment]
> the potential use of different vehicles is why I originally put capacity in GTFS-RT rather than trips.txt. If we need to allow realtime overrides - is it still useful to have it in the GTFS trips.txt?

I don’t see the value in having it in the static. While a particular vehicle’s capacity is static, a trip’s isn’t. And even if it was I don’t see much value for a trip planner because it’s the real-time load that matters. 

I think of load certainly as a real-time thing, and “the capacity of the vehicle running this trip currently” makes more sense as a RT aspect of a trip than a Static one to me — until Vehicles is widely adopted and could be referenced.

**Comment by mike-swiftly** [comment]
Vehicle capacity could perhaps be included in static info, but definitely not in trips.txt since as others have stated capacity varies. It is not static with respect to trips. 

But vehicle capacity could potentially be supplied in a separate and new GTFS file, perhaps called vehicles.txt. That info could include not just passenger capacity but other amenities such as bike racks, USB charging, WiFi, etc. I think this could be a good change to consider, but should be considered separately from occupancy status.

**Comment by botanize** [comment]
I strongly prefer `OccupancyStatus` if there must be only one. But I think the two options have different uses, and should co-exist. 

I believe `occupancy_percentage` causes confusion about what the occupancy is relative to and suffers from a false sense of precision. Furthermore, I think that the many levels of the enum provide agencies a way to best communicate loads as they see fit, whether that's using all levels the way CapMetro plans to, or choosing the three or four that meet their needs. I do not think consumers should attempt to translate `OccupancyStatus` to a percentage for display, as that changes the meaning of the field.

People don't know how many seats are on our vehicles, and may not be aware of current seating restrictions, so 80% means nothing, or it's confusing because they don't know if it's 80% of the max capacity or COVID-19 capacity. You might think 80% means 8 people, you might think it means 36 people or even 48 people (if you think of capacity as COVID restricted 10, seated capacity of 40 or standing capacity of 60 people on a 40 foot bus). Depending on your interpretation you'd think there was room for 2, 24 or 12 people, hardly actionable information.

`occupancy_percentage` also suffers from implied precision. Our uncertainty about realtime load is high, so even if we do implement `occupancy_percentage` we'd use large bands of 20 points or more. But if you see 80%, you don't know that we only report in 20 point increments. 

I think the descriptive messages are much more clear because they aren't in the context of an unknown and possibly changing standard. I actually think "FULL" and "few seats available" are perfect descriptions of what the current vehicle load means to *me*. It's actionable information. Full means there's no space for *me*, whether that's because the bus can't fit another person or because there's social distancing restrictions on space. Few seats available means few seats available to *me*, regardless of how many are empty, I don't need to know the capacity of the vehicle, or what restrictions are in place, because the descriptive text tells me approximately what's available to *me*, it's not relative to some unknown or changing standard the way `occupancy_percentage` is relative to either the seating capacity, crush load capacity or restricted capacity of a vehicle whose capacity I don't know anyway.

**Comment by antrim** [comment]
Hi all - 

I’ve attempted to summarize the positions in this thread and draw a few conclusions. Please review and respond!

#### Would prefer `OccupancyStatus` (if we chose one field):
- @skinkie (Stefan de Konink)
- @botanize (Joey Reid, Minneapolis Metro)
- @juanborre

#### Would prefer `occupancy_percentage` (if we chose one field):
- @stevenmwhite (Steven White, @syncromatics)

#### 4 people said the fields serve separate purposes and should not be mutually exclusive:
- @stevenmwhite (Steven White - Syncromatics)
- @mike-swiftly (Mike Smith - Swiftly)
- @mcossette1 
- @BodoMinea (Bodo Minea)

Note there was recently strong support for adding `occupancy_percentage` as an experimental field (#213).

#### Reasons in favor of `occupancy_percentage`:
- Lends itself to situational adjustments (e.g. change in maximum occupancy policy)
- Gives consumers flexibility in how it is displayed
- More granularity (this was the argument for adding `occupancy_percentage` #213)
- Some messages “Standing Room Only” can be difficult to assert with confidence. What if the bus is quite full but some seats are still available because passengers are standing?

#### Reasons in favor of OccupancyStatus:
- Compatibility with SIRI
- Doesn’t provide unnecessary or false precision
- Provides enough information: Producer can specify if a passenger is likely to find a seat
“Few seats available,” “many seats available,” “standing room only” and “full” are clear messages for a customer

#### Current use in consuming applications
Both Transit and Google Maps are using a combination of messages and icons to show occupancy [below]. Would `occupancy_percentage` be more useful to determine what icons to show:

_Google Maps in Sydney_ - People icon:
![Google Maps in sydney](https://user-images.githubusercontent.com/6495639/82377508-9f21ce80-99d8-11ea-9d71-4b08a19a9ba9.png)

_Transit in NYC_ - Subtle green shading indicates occupancy:
![Transit in NYC](https://user-images.githubusercontent.com/6495639/82377492-97622a00-99d8-11ea-9be5-9230418a8008.png)

#### Concept: Adding capacity information
Matt Caywood (@caywood ) started a discussion of how to describe vehicle capacity to give more context for occupancy data. If occupancy was in the context of capacity (both seating + standing), that would give applications a lot of flexibility for presentation in the UI, but that would be a lot to add to a transit data pipeline. Perhaps this is something to add over time, in coordination with static extensions like [GTFS-Vehicles](http://bit.ly/gtfs-vehicles)?

#### Here are some conclusions
Please post if you see holes in these (or post if you agree and they make sense to you 😀 ).
- Most people who responded said that `OccupancyStatus` and `occupancy_percentage` are not mutually exclusive. If we agree on that, then users of the spec will need to settle on what separate functions they serve. (Below are some ideas)
- `OccupancyStatus` is useful to show messages to passengers, without making a precise statement about the occupancy of the vehicle. If we all agree, then we should develop a common understanding and decide if:
  - Some messages should be deprecated or if all messages continue to be allowed (say, SIRI enumerations: SEATS AVAILABLE, STANDING ROOM ONLY, FULL)
  - Any new messages are added or if free text is allowed in the future
- `occupancy_percentage` is most useful right now for showing occupancy via an icon, which some applications currently do.
- If capacity is included with `occupancy_percentage` that will open up new opportunities for consuming applications. The use cases (app implementations) are not yet clear but may emerge over time.

**Comment by caywood** [comment]
Thanks @antrim for a solid summary. 

Please don't even consider consumers in this decision process, we will be fine - companies like TransitScreen (and even Google) can move as fast as necessary once the data exists.

For me, the paramount concern has always been **producers** - particularly the 99% of agencies who currently don't have any occupancy data - we need them to start reporting _something_ if this is going to impact the present transit crisis. So we really need to hear from agencies who are not yet producing data, or are currently implementing it.

**passenger_count_estimate** has the advantage that it's very straightforward to implement provided the correct data is in the pipeline (like APCs and vehicle capacities) - this is I think the same data you would need to produce either OccupancyStatus or occupancy_percentage, just in the most straightforward format possible. 

**occupancy_percentage** is pretty straightforward assuming we specify it's a percentage of the maximum occupancy per current agency regulations. (However if that changes over time, that creates a minor burden for producers.)

The complexity of **OccupancyStatus** is its downfall - every agency is going to have to have meetings to figure out what "Standing Room Only" means in COVID, and normally, etc.. Compatibility with SIRI is nice to have but not at the expense of getting things done at a critical time.

**Comment by stevenmwhite** [comment]
@antrim, I would second @caywood's thanks for the great summary, as well as his focus on producers (I am a producer so take that with a grain of salt, if you will -- but I'm also a consumer [mostly of feeds I produce]).

I think your conclusions are all reasonable and accurate.

I think Percentage is the most useful of them all, and you've clarified how Status can be useful as well (in particular, "to show messages to passengers"). Further clarifying those messages is a good idea and while using SIRI as a model (rather an always trying to reinvent the wheel) is a good idea, I wouldn't say that compatibility with SIRI should be a goal on its own. That being said, I would be in favor of a smaller set of enumerations as you've listed.

**Comment by esteveavi** [comment]
Hi, in this COVID de-escalation phase authorities are recommending a 50% max occupation per trip. How would you solve that using OccupancyStatus enumeration? 
By using occupancy_percentage it seems easy.
Thank you.

**Comment by antrim** [comment]
@esteveavi: Under the framework above, operators would use OccupancyStatus messages that are meaningful to passengers during Covid: `SEATS_AVAILABLE` and `FULL`. Seat availability would be according to current operating rules.

Question: Is/would `FEW SEATS AVAILABLE` and `MANY SEATS AVAILABLE` be a useful and meaningful distinction for passengers?

**Comment by esteveavi** [comment]
Thanks, @antrim.  In this de-escalation, a 50% occupation with FEW_SEATS_AVAILABLE can be considered full.

**Comment by willwong430** [comment]
The MTA likes the idea of using occupancy_percentage as the primary indicator for our realtime APC feed. We should be able to easily adjust our enumeration according to what the community deems most effective in defining the degree of passenger occupancy.

**Comment by darylweinberg** [comment]
CapMetro thinks there are uses for both fields. If we had to pick one it would be Occupancy Status but would prefer to have both available.

We agree with reducing the number of enumerated values but at the moment we are working with our consuming partners to do that when displaying to customers, not on the back end where it is produced.

Thank you,

[http://www.capmetro.org/email/eSigLogo.png]<http://www.capmetro.org/>
Daryl Weinberg
Transit Systems Architect
[http://www.capmetro.org/email/sig_fb.png]<https://www.facebook.com/capitalmetro?_rdr=p>  [http://www.capmetro.org/email/sig_tw.png] <https://twitter.com/CapMetroATX>
o: 512-369-6216
|
e: Daryl.Weinberg@capmetro.org<mailto:Daryl.Weinberg@capmetro.org>
|
w: capmetro.org<http://www.capmetro.org/>

**Comment by esteveavi** [comment]
In TMB (Barcelona, Spain) we are using occupancy_percentage to show occupation:
Here is an example: https://www.tmb.cat/en/barcelona/metro/-/lineametro/L5
We are using 5 levels now to test how it works.

**Comment by antrim** [comment]
@esteveavi : Since it appears TBM is expressing average historical occupancy (rather than real-time), one resource you may find useful is a draft spec called GTFS-Occupancies (http://bit.ly/gtfs-occupancies) which MobilityData circulated last year. To my knowledge it has not been implemented. We would be interested in your comments. Note that the current draft expresses average occupancy on a trip/stop level, not a route/stop level.

**Comment by paulswartz** [comment]
After a lot of internal discussion, @mbta is going with 3 occupancy buckets in `occupancy_status` instead of `occupancy_percentage`. The exact buckets, and what status they map to, are still being discussed.

**Comment by Dave-TfNSW** [comment]
I think Transport for NSW would prefer to stick occupancy status rather than percentage, as it is “less precise and more accurate” as has been stated in the thread, and allows for clearer customer messaging. We do not see them as mutually exclusive however. It may even be helpful to have both for communicating changing COVID thresholds. For example we recently changed our highest level for buses from 100% of seats to 31%, but consumers would only know this from our direct communications or scrutiny of our documentation.

**Comment by hfonbouze** [comment]
Here’s what we think at Ineo Systrans (supplier of CAD-AVL systems).
The percentage is not easily understandable for the end user (passenger), especially for high values (what’s mostly unclear is : percentage of what?). It has to be translated into a category of riding condition (i.e. a status).
The information should also be consistent between the different channels. It may use different renderings but the meaning has to be the same. Which applies to apps using GTFS-RT but also to information given through other means (displays at stop, dispatchers’ screens…).
Therefore the translation (from percentage to status) should be done as upstream as possible, which means at producer’s level.
For these reasons, we (as producers) prefer to use the occupancy_status.

**Comment by skinkie** [comment]
@hfonbouze in de past it has been discussed frequently that some CAD-AVL systems in the USA are using GTFS-RT as backhaul information exchange. What is the value or domain that your implementation uses from vehicle to dispatch?

**Comment by hfonbouze** [comment]
@skinkie Our standard system (which includes both vehicle equipments and dispatch) doesn't use GTFS-RT for internal exchanges (but a proprietary interface).

**Comment by skinkie** [comment]
> @skinkie Our standard system (which includes both vehicle equipments and dispatch) doesn't use GTFS-RT for internal exchanges (but a proprietary interface).

Still the same question applies, are you using discrete values, an enumeration, or a percentage in this proprietary interface.

**Comment by CleverDevices-gtfs** [comment]
We had an internal discussion about this at Clever Devices.  We believe just using Occupancy Status is best because:
•	As a GTFS-RT producer, the occupancy status allows an agency’s policy to communicate  how full the vehicle is instead of through raw numbers. 
•	A load percentage may be confusing for the customer and could vary based on the type of vehicle assigned to the route.  APC sensor quality and type will affect how accurately the load is reported; in ridership reporting packages this quality is handled by sampling many boardings over a statistically significant number of trips  and discarding outlier trips and blocks, which can not be done in real time and there is no extra data in the standard to account for variability.

**Comment by osmaa** [comment]
Has anyone considered the question of encoding occupancy on a train car / compartment level? The simplest approach would be to extend OccupancyStatus to a many cardinality field, though that would leave it to convention to interpret ordering etc.

**Comment by skinkie** [comment]
> Has anyone considered the question of encoding occupancy on a train car / compartment level? The simplest approach would be to extend OccupancyStatus to a many cardinality field, though that would leave it to convention to interpret ordering etc.

https://github.com/google/transit/issues/223#issuecomment-625498601

**Comment by barbeau** [comment]
>Has anyone considered the question of encoding occupancy on a train car / compartment level? The simplest approach would be to extend OccupancyStatus to a many cardinality field, though that would leave it to convention to interpret ordering etc.

@osmaa See the draft [GTFS-Vehicles spec](http://bit.ly/gtfs-vehicles) for an thorough example of this.

**Comment by kronster** [comment]
There is higher value in the enumeration.  Showing the percentage unnecessarily exposes rider safety concerns.  The same applies to the public display of the block_id with driver safety.

**Comment by abyrd** [comment]
> > Here is another topic for discussion with respect to occupancyStatus and occupancy_percentage: should they also be included with predictions in the TripUpdates feed?
> 
> @mike-swiftly I'd prefer to have TripUpdates specifically contain **predictive** occupancy information within StopTimeUpdates. As you imply, if an agency knows historical on/offs at specific stops, the current occupancy could likely differ significantly from the expected occupancy in a few stops (e.g., stops near the end of the line).

I was just looking at a VehiclePosition message today and had the same thought: why is occupancy information in VehiclePositions instead of TripUpdates? @barbeau your framing makes sense: TripUpdates could be used for predictive information and VehiclePositions for current information. But if that is the reasoning behind putting occupancy in VehiclePositions,  it should be clearly stated in documentation and best practices guides, as this distinction is not obvious from the names of these message types. In fact documentation specifically says VehiclePositions are for communicating the geographic location of vehicles.

Even if TripUpdates contained occupancy predicted at each StopTimeUpdate as @barbeau suggests, this would still be capable of communicating current occupancy in the same way it communicates current on-time status. Note also that TripUpdate has a delay field at the top level outside StopTimeUpdates giving "the current schedule deviation for the trip." It seems completely reasonable to include current occupancy information in the same way in a top level field.

**Comment by barbeau** [comment]
>But if that is the reasoning behind putting occupancy in VehiclePositions, it should be clearly stated in documentation and best practices guides, as this distinction is not obvious from the names of these message types. In fact documentation specifically says VehiclePositions are for communicating the geographic location of vehicles.

I agree 👍 

>Note also that TripUpdate has a delay field at the top level outside StopTimeUpdates giving "the current schedule deviation for the trip." It seems completely reasonable to include current occupancy information in the same way in a top level field.

I would actually like to see `TripUpdate.delay` field redefined to being a prediction and a new field `VehiclePosition.schedule_deviation` (or some variant of this) be clearly defined as observed schedule deviation and not a prediction. I've seen producers treat `TripUpdate.delay` the same way as `StopTimeUpdate.delay` - as a prediction at the trip level, if the prediction engine isn't capable of per-stop predictions. And to be fair the spec is not clear about the distinction between schedule deviation and a prediction - it doesn't define the concept, and with the field name being the same it's easy to see how producers can assume the same semantic definition applies to both fields. IMHO redefining/moving the field would be more in-line with how VehiclePosition and TripUpdate should be delineated - VehiclePosition is an observed fact about vehicle progress, with TripUpdates being predicted values.

There have also been recent discussion of the concept of observed trip progress being added to feeds as well, which would also fit into the above structure - something like `VehiclePosition.VehicleTripStatus` that would be similar to the existing `VehiclePosition.VehicleStopStatus` field that tracks a vehicle's observed progress in relation to the trip instead of the stop, but I'll open a separate issue to discuss that.

In summary, I'd still prefer that observed occupancy be contained in VehiclePositions, with predicted occupancy be contained in TripUpdates.

**Comment by mike-swiftly** [comment]
I'm in strong agreement with the idea that observed occupancy be contained
in VehiclePositions with predicted occupancy be contained in TripUpdates.
But I wanted to state another reason why these should be separate. Key
issue is that observed occupancy, though complicated, is much simpler to
implement than predicted occupancy. By clearly separating these issues we
can continue to make immediate progress with observed occupancy and then
slowly make progress with predicted occupancy.

*Michael Smith*
*CIO, **Swiftly, Inc*
(415) 260-4700

**Comment by scmcca** [comment]
Hi everyone. 

As you may be aware from conversations in #212 and the [GTFS Realtime mailing list](https://groups.google.com/g/gtfs-realtime/c/FA2_5_pOxj0), #259 recently passed on clarifying the use of OccupancyStatus. Part of this clarification was to explicit the separate uses of OccupancyStatus and `occupancy_percentage` for reasons echoed throughout this thread around describing actual occupancy states and describing linear scales. 

Considering #259 and the discussions here, I think it is safe to say the conversation is at least no longer about OccupancyStatus "vs" `occupancy_percentage`, as both serve separate purposes.

One of the next items to tackle is the official adoption of an occupancy description in GTFS Realtime. As both OccupancyStatus and `occupancy_percentage` are experimental, they are blocking an attempt to synchronize occupancy descriptions for static historical/predicted occupancies in proposal #240. On top of this, OccupancyStatus has been experimental for 6+ years, well past the 2-year experimental period outlined in the [specification amendment process](https://github.com/google/transit/blob/master/gtfs-realtime/CHANGES.md#experimental-fields), demonstrating that it is worthwhile to the community. I see a few options amongst many:
1. Officially adopt the original 7 OccupancyStatus enum values: `EMPTY`,`MANY_SEATS_AVAILABLE`, `FEW_SEATS_AVAILABLE`, `STANDING_ROOM_ONLY`, `CRUSHED_STANDING_ROOM_ONLY`, `FULL`, `NOT_ACCEPTING_PASSENGERS`. This would avoid conflicts with producers and consumers who have been historically using some or all of these values.
2. Increase endorsement and officially adopt `occupancy_percentage`. Not as widely used as OccupancyStatus, but allows direct occupancy descriptions on a linear scale giving producers whatever granularity they want and giving freedom to consumers for how they want to display the data (icons, buckets, etc).
3. Open a separate proposal that uses enum values on a linear scale: `NOT_CROWDED`, `SOME_CROWDING`, `CROWDED` (cc: #212). This would address previous proposals to describe occupancies on a linear scale using enum values while avoiding conflicts with historical OccupancyStatus users, but would rigidly define the buckets to use (the community would have to settle on a number whether its 3, 4, or more).

What are everyone's thoughts on the options (or combination of options) above? Thanks.

**Comment by gcamp** [comment]
3. is the one that is the most pressing and that makes the most sense to me.

**Comment by RTL-Gilles-Ferland** [comment]
Hello, option 3 for us as well. Thanks.

**Comment by CleverDevices-gtfs** [comment]
Clever Devices’ stance is that only Occupancy Status is provided as part of GTFS-RT, and that Option 3 with three crowding enumerations be the option that is advanced. 
•	The occupancy status should allow a Transit Agency to communicate crowding in meaningful terms for use in journey-planning by their riders.
•	Using a percentage of vehicle capacity occupied may be confusing for the customer and made even more confusing for agencies with a diverse fleet and different vehicle capacities. 
•	APC sensor quality and type will affect how accurately the load is reported and it is our belief that this level of precision is not appropriate for real-time use and may decay ridership trust in the metric when the measures do not align precisely with real-time conditions. This uncertainty is addressed in ridership reporting packages by large sample size and post-hoc business rules to handle outliers and erroneous data. This cannot be done in real-time.  
•	For these reasons, we also believe that the number of enumerations should be limited to three. These broad measures allow an agency to communicate crowding consistent with their own policies in ways that can be easily understood and consumed by passengers without overstating the statistical confidence of real-time passenger counts. The proposed measures NOT_CROWDED, SOME_CROWDING, and CROWDED best capture the need to communicate crowding in real-time consistent with the limits of APC technology.
•	No limits should be placed on passenger counts or % passenger occupancy that align with each of the above measures. This allows for Transit Agencies the flexibility to define their own policy and allow it to evolve over time. Communicating crowding may currently be needed for social distancing and public health reasons. In the future GTFS-RT producers may use this as a measure that helps define passenger comfort in transit vehicles and should have the flexibility to do so.

**Comment by scmcca** [comment]
Thanks for the feedback! As per the comments above, I opened a separate PR for CrowdLevel at #261. Feel free to engage over there to provide any feedback on the proposal.
